create_infotexts.py

```diff
@@ -239,12 +216,57 @@ def __init__(self):
         self.content_cats = []  # content cateogories without 'Category:'-prefix
         self.meta_cats = []  # maintance categories without 'Category:'-prefix
         self.data = {}  # dictionary holding individual field values as wikitext
+        self.filename = ""  # without filename extension
 
         batch_cats = ["Swedish Cyprus Expedition",
                       "Media_contributed_by_SMVK_2017-02"]
 
         self.meta_cats.extend(batch_cats)
 
+    def create_commons_filename(self, item):
+        """
+        Transform original filename into Wikimedia commons style filename.
+
+        See https://phabricator.wikimedia.org/T156612 for definition.
+        Note: the returned filename does not include file extension e.g. '.tif'
+
+        :param item: dictionary conatining metadata for one image
+        :return: None (populates self.filename)
+        """
+        fname_str = ""
+        fname_str += re.sub(" Svenska Cypernexpeditionen\.?", "", item["Beskrivning"])
+        if not fname_str.endswith(" "):
+            fname_str += " "
+
+        # Enrich with keywords
+        if self.data["keyword_list"]:
+            for keyword in self.data["keyword_list"]:
+                if keyword not in item["Beskrivning"]:
+                    fname_str += keyword + " "
+
+        # Enrich with regional data
+        if not item["Ort, foto"] in item["Beskrivning"]:
+            fname_str += item["Ort, foto"] + " "
+
+        if not item["Region, foto"] in item["Beskrivning"]:
+            fname_str += item["Region, foto"] + " "
+
+        if not item["Land, foto"] in item["Beskrivning"]:
+            fname_str += item["Land, foto"] + " "
+
+        # Ensure first descriptive part is not empty
+        if fname_str == " ":
```

Comment:
You can replace this by:
```python
if not fname_str.strip():
    fname_str = "Svenska Cypernexpeditionen 1927-1931"
```
or (since you probably want to strip anyway)
```python
fname_str = fname_str.strip()
fname_str = fname_str or "Svenska Cypernexpeditionen 1927-1931"
```